File: code/cpp/974.cpp
//https://leetcode.com/problems/subarray-sums-divisible-by-k/
int subarraysDivByK(vector<int>& nums, int k) {
	int sum = 0, rem = 0, count = 0;
	unordered_map<int, int> mp;
	mp[0] = 1;
	for (int n : nums) {
		sum += n;
		rem = (sum % k + k) % k;
		count += mp[rem]++;
	}
	return count;
}

File: code/python/974.py
#https://leetcode.com/problems/subarray-sums-divisible-by-k/
def subarraysDivByK(nums: List[int], k: int) -> int:
    sum = rem = res = 0
    mp = {0: 1}
    for num in nums:
        sum += num
        rem = (sum % k + k) % k
        if rem in mp:
            res += mp[rem]
            mp[rem] += 1
        else:
            mp[rem] = 1
    return res
